index.js

```diff
@@ -30,16 +30,34 @@ module.exports = function generateRequireForUserCode (scopedDirs, options) {
     }
   })
 
-  function deleteModuleFromCache (m) {
+  function isSubPath (parent, modulePath) {
+    const relative = path.relative(parent, modulePath)
+    return relative && !relative.startsWith('..')
+  }
+
+  function shouldDeleteFromCache (modulePath, directoriesToClear) {
+    if (_.isEmpty(directoriesToClear) || modulePath === baseModule.id) {
+      return true
+    }
+
+    return _.some(directoriesToClear, (directoryToClear) => isSubPath(directoryToClear, modulePath))
+  }
+
+  function deleteModuleFromCache (m, directoriesToClear) {
     if (m && m.id && m.id.endsWith('.node')) {
       m.parent = null
       return
```

Comment:
Should we return `false` for consistency with the other paths returning a boolean?

Reply:
https://github.com/wix/scoped-require/pull/11/commits/b257d3a7678e195eed7cee95d6e4ed0459745f62